firebase-firestore/src/main/java/com/google/firebase/firestore/local/BundleSerializer.java

```diff
@@ -0,0 +1,350 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import android.util.Base64;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+
+import com.google.firebase.Timestamp;
+import com.google.firebase.firestore.core.Bound;
+import com.google.firebase.firestore.core.FieldFilter;
+import com.google.firebase.firestore.core.Filter;
+import com.google.firebase.firestore.core.OrderBy;
+import com.google.firebase.firestore.core.Query;
+import com.google.firebase.firestore.model.Document;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.model.FieldPath;
+import com.google.firebase.firestore.model.ObjectValue;
+import com.google.firebase.firestore.model.ResourcePath;
+import com.google.firebase.firestore.model.SnapshotVersion;
+import com.google.firebase.firestore.model.Values;
+import com.google.firebase.firestore.remote.RemoteSerializer;
+import com.google.firestore.v1.ArrayValue;
+import com.google.firestore.v1.MapValue;
+import com.google.firestore.v1.Value;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.NullValue;
+import com.google.type.LatLng;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/** A JSON serializer to deserialize Firestore Bundles. */
+/* package= */ class BundleSerializer {
+
+  private final RemoteSerializer remoteSerializer;
+
+  public BundleSerializer(RemoteSerializer remoteSerializer) {
+    this.remoteSerializer = remoteSerializer;
+  }
+
+  public NamedQuery fromNamedQuery(String json) throws JSONException {
```

Comment:
This class is mixing "from" and "decode" prefixes for decoding methods. The other android serializers use "decode" so maybe stay consistent with that?

(We do have static factory methods with "from" names, but these work well because the class name gets prefixed: `DocumentKey.fromName` reads clearly, but just `jsonSerializer.fromName` would not.)

Reply:
Renamed all methods to use `decode` prefix.